port/zephyr/Kconfig

```diff
@@ -6,6 +6,16 @@ menuconfig GOLIOTH_FIRMWARE_SDK
 	select POSIX_API # libcoap
 	select REBOOT if BOOTLOADER_MCUBOOT
 	select REQUIRES_FULL_LIBC # cJSON, libcoap
+	select MBEDTLS
+	select MBEDTLS_DTLS if MBEDTLS_BUILTIN
+	select NET_SOCKETS
+	select NET_UDP
+	select ZCBOR
+	imply DNS_RESOLVER if NET_NATIVE
+	imply NET_SOCKETS_SOCKOPT_TLS
+	imply NET_SOCKETS_ENABLE_DTLS
+	help
+	  Enable library for communication with Golioth cloud.
```

Comment:
Maybe let's remove those symbols from `examples/zephyr/golioth_basics` and actually see how much this simplifies `prj.conf`.

Reply:
I have done so by adding this commit:
https://github.com/golioth/golioth-firmware-sdk/pull/129/commits/b9048cbcef4ca8fed4dd91ac6b80a2681cffc552

I was trying to keep golioth_basics as a more verbose example. Combining the default symbols from this commit with the common/Kconfig.defconfig committed later in this PR makes for nearly-empty prj.conf files in the more simplified samples.

Reply:
There should be really just a single way how to configure examples. Otherwise such inconsistency is confusing to people wanting to compare those samples (e.g. "what does it take to configure more services at once, just like in golioth_basics sample").

Reply:
I discussed with @sam-golioth and have left these symbols as is for now.